core/math/src/main/java/org/arakhne/afc/math/geometry/d2/afp/BasicPathShadow2afp.java

```diff
@@ -209,83 +207,36 @@ private void discretizePathIterator(
             case LINE_TO:
                 endx = element.getToX();
                 endy = element.getToY();
-                crossSegmentTwoShadowLines(
-                        curx, cury,
-                        endx, endy,
-                        x1, y1, x2, y2);
-                if (this.crossings == MathConstants.SHAPE_INTERSECTS) {
-                    return;
-                }
+                setLineToFromDiscretizePathIterator(curx, cury, endx, endy, x1, y1, x2, y2);
                 curx = endx;
                 cury = endy;
                 break;
             case QUAD_TO:
                 endx = element.getToX();
                 endy = element.getToY();
                 // only for local use.
-                localPath = pi.getGeomFactory().newPath(pi.getWindingRule());
-                localPath.moveTo(curx, cury);
-                localPath.quadTo(
-                        element.getCtrlX1(), element.getCtrlY1(),
-                        endx, endy);
-                discretizePathIterator(
-                        localPath.getPathIterator(MathConstants.SPLINE_APPROXIMATION_RATIO),
-                        x1, y1, x2, y2);
-                if (this.crossings == MathConstants.SHAPE_INTERSECTS) {
-                    return;
-                }
+                setQuadToFromDiscretizePathIterator(element, pi, x1, y1, x2, y2, curx, cury, endx, endy);
```

Comment:
The replaced code was supposed to exit this function if crossings is not zero. The replacement function does not provide the same feature.